### ISSUE TITLE
Bug in hessian structure

### ISSUE THREAD

**Issue by tmigot** [JuliaSmoothOptimizers/NLPModelsJuMP.jl#143]
Oh, I think the problem comes from the model actually. I did the following (a bit nervous) test:
```
julia> using NLSProblems, NLPModels

julia> nlp = hs06()

julia> hess_structure(nlp)
([1, 267043088], [1, 167116808])

julia> hess_structure(nlp)
([1, 331019536], [1, 269359968])

julia> hess_structure(nlp)
([1, 331019536], [1, 167149984])

julia> hess_structure(nlp)
([1, 331019536], [1, 167150048])

julia> hess_structure(nlp)
([1, 5124065472], [1, 5124065472])
```

_Originally posted by @tmigot in https://github.com/JuliaSmoothOptimizers/CaNNOLeS.jl/issues/43#issuecomment-1308778149_

**Comment by amontoison** [comment]
@tmigot 
I found the bug when I worked on the support of quadratic constraints.
I opened a PR to fix it: #185 
I will tag a new release after.

**Comment by tmigot** [comment]
Great! Thanks
